src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -821,6 +821,15 @@ private static Expression CreateArgument(ParameterInfo parameter, RequestDelegat
                     $"Nested {nameof(AsParametersAttribute)} is not supported and should be used only for handler parameters.");
             }
 
+            // Parameters like IFormFileCollection should throw "The abstract type ... is not supported" rather than
```

Comment:
Is there a reason we can't throw this exception in the same place where the other exception is thrown?

https://github.com/dotnet/aspnetcore/blob/5f156a2449edaf10767a4f358af51785a1015d3f/src/Shared/ParameterBindingMethodCache.cs#L363

Reply:
@captainsafia 
Fixed. I moved the exception to ParameterBindingMethodCache.cs, works well.